src/zenml/artifacts/utils.py

```diff
@@ -272,29 +245,113 @@ def _create_version() -> Optional[ArtifactVersionResponse]:
         )
 
     if manual_save:
+        _link_artifact_version_to_model_version_in_context(
+            name=name,
+            response=response,
+            is_model_artifact=is_model_artifact,
+            is_deployment_artifact=is_deployment_artifact,
+        )
+
+    return response
+
+
+def link_existing_data_as_artifact(
+    folder_or_file_uri: str,
+    name: str,
+    version: Optional[Union[int, str]] = None,
+    tags: Optional[List[str]] = None,
+    has_custom_name: bool = True,
+    is_model_artifact: bool = False,
+    is_deployment_artifact: bool = False,
+    artifact_metadata: Dict[str, "MetadataType"] = {},
+) -> "ArtifactVersionResponse":
+    """Link existing data stored in the artifact store as an artifact.
+
+    Args:
+        folder_or_file_uri: The full URI within the artifact store to folder or a file.
+        name: The name of the artifact.
+        version: The version of the artifact. If not provided, a new
+            auto-incremented version will be used.
+        tags: Tags to associate with the artifact.
+        has_custom_name: If the artifact name is custom and should be listed in
+            the dashboard "Artifacts" tab.
+        is_model_artifact: If the artifact is a model artifact.
+        is_deployment_artifact: If the artifact is a deployment artifact.
+        artifact_metadata: Metadata dictionary to attach to the artifact version.
+
+    Returns:
+        The saved artifact response.
+
+    Raises:
+        FileNotFoundError: If the folder URI is outside of the artifact store
+            bounds.
+    """
+    client = Client()
+
+    # Get the current artifact store
+    artifact_store = client.active_stack.artifact_store
+
+    if not folder_or_file_uri.startswith(artifact_store.path):
+        raise FileNotFoundError(
+            f"Folder `{folder_or_file_uri}` is outside of "
+            f"artifact store bounds `{artifact_store.path}`"
+        )
+
+    _check_if_artifact_with_given_uri_already_registered(
+        artifact_store=artifact_store,
+        uri=folder_or_file_uri,
+        name=name,
+    )
+
+    artifact = _get_or_create_artifact(
+        name=name,
+        has_custom_name=has_custom_name,
+        tags=tags,
+    )
+
+    # Create the artifact version
+    def _create_version(
+        version: Union[int, str],
+    ) -> Optional[ArtifactVersionResponse]:
+        artifact_version = ArtifactVersionRequest(
+            artifact_id=artifact.id,
+            version=version,
+            tags=tags,
+            type=ArtifactType.DATA,
+            uri=folder_or_file_uri,
+            materializer=source_utils.resolve(PreexistingDataMaterializer),
+            data_type=source_utils.resolve(Path),
```

Comment:
This should be `PreexistingArtifactPath` I think?

Reply:
Yeah, thanks for the catch